src/main/java/com/carservice/manager/controller/VehicleController.java

```diff
@@ -25,35 +29,53 @@ public String test() {
 
     @GetMapping("/cars")
     public String carsView(Model model) {
-        model.addAttribute("cars", vehicleService.getVehicleModels());
+        model.addAttribute("cars", vehicleService.getVehicleModels("main"));
+        model.addAttribute("time", true);
         return "cars";
     }
 
     @GetMapping("/add")
-    public String addCar(Model model) {
-        model.addAttribute("car", new VehicleModel());
+    public String addCar(Model model, VehicleModel vehicleModel) {
+        model.addAttribute("vehicleModel", vehicleModel);
         return "add";
     }
 
     @PostMapping("/add")
-    public String addCar(VehicleModel vehicleModel) {
-        vehicleService.addVehicle(vehicleModel);
+    public String addCar(@Valid VehicleModel vehicleModel, BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            System.err.println(bindingResult.getAllErrors().stream().map(ObjectError::toString).collect(Collectors.joining("\n")));
+            return "add";
+        }
+        vehicleService.addVehicle(vehicleModel, "main");
         return "redirect:/cars";
     }
 
     @PostMapping("/repair")
     public String repairCar(@RequestParam("id") int id) {
```

Comment:
If there is only a request parameter (id) and there is no request body, then maybe it can be GET request instead of POST? 🤔